### PR TITLE
fix #1075 Remove REMOTE_ADDRESS tag from the metrics reported by Micr…

### LINKED ISSUE #1075: Server metrics cardinality explosion
## Expected Behavior
A reasonable amount of series for every reactor_netty_http_server_* metric.

## Actual Behavior
There is a cardinality explosion with one per `client ip` * `client random port` * `other tags`

metrics-cardinality

## Steps to Reproduce
- Enable HttpServer.metrics(true)
- Curl some endpoints

## Possible Solution
Remove remote_address tag from all server metrics

## Your Environment

* Reactor version(s) used: 0.9.4
* Other relevant libraries versions (eg. `netty`, ...): Spring Boot 2.2.4.RELEASE
* JVM version (`javar -version`): OpenJDK 11
* OS and version (eg `uname -a`): MacOS Catalina

### PR DESCRIPTION
**Pull request by violetagg**
…ometerHttpServerMetricsRecorder

MicrometerHttpServerMetricsRecorder does not record metrics on protocol level

Fixes #1075

### CODE REVIEW

File: src/main/java/reactor/netty/http/MicrometerHttpMetricsRecorder.java [hunk 1]
```diff
@@ -19,28 +19,23 @@
...
-import static reactor.netty.Metrics.REMOTE_ADDRESS;
 import static reactor.netty.Metrics.RESPONSE_TIME;
-import static reactor.netty.Metrics.URI;
 
 /**
  * @author Violeta Georgieva
  * @since 0.9
  */
-public class MicrometerHttpMetricsRecorder extends MicrometerChannelMetricsRecorder implements HttpMetricsRecorder {
+public abstract class MicrometerHttpMetricsRecorder extends MicrometerChannelMetricsRecorder implements HttpMetricsRecorder {
```
Comment (simonbasle) [inline src/main/java/reactor/netty/http/MicrometerHttpMetricsRecorder.java:38]: I _think_ that is binary compatible, the `abstract` + removal of overrides in this base class, but can you confirm JAPICMP doesn't find any issue here?

File: src/main/java/reactor/netty/http/server/HttpServerBind.java [hunk 2]
```diff
@@ -380,6 +380,10 @@ public void accept(ConnectionObserver listener, Channel channel) {
 				if (channelMetricsRecorder instanceof HttpServerMetricsRecorder) {
 					p.addAfter(NettyPipeline.HttpTrafficHandler, NettyPipeline.HttpMetricsHandler,
 							new HttpServerMetricsHandler((HttpServerMetricsRecorder) channelMetricsRecorder, uriTagValue));
+					if (channelMetricsRecorder instanceof MicrometerHttpServerMetricsRecorder) {
+						// MicrometerHttpServerMetricsRecorder does not implement metrics on protocol level
+						p.remove(handler);
```
Comment (simonbasle) [inline src/main/java/reactor/netty/http/server/HttpServerBind.java:385]: nitpick: reading the code it was not immediately clear `p` was the channel-level metrics and was different from the `HttpServerMetricsHandler` (applies to 3 sites)

File: src/main/java/reactor/netty/http/MicrometerHttpMetricsRecorder.java (hunk 1 above)
Comment (violetagg) [reply src/main/java/reactor/netty/http/MicrometerHttpMetricsRecorder.java]: There is an issue if this class is inherited.
Fixed

File: src/main/java/reactor/netty/http/server/HttpServerBind.java (hunk 2 above)
Comment (violetagg) [reply src/main/java/reactor/netty/http/server/HttpServerBind.java]: fixed

**Pull request merged**